### PR TITLE
Update docs and README for XGBoost explain_prediction

### PR DESCRIPTION
**Pull request by lopuhin**
Continued from #117 

- Docs: update README, overview, xgboost.rst (added a link to the blogpost in docstring) - maybe I missed some other place in the docs?
- Make most functions in eli5.xgboost private
- Update XOR caveat

### CODE REVIEW

File: docs/source/autodocs/xgboost.rst [hunk 1]
```diff
@@ -2,7 +2,10 @@ eli5.xgboost
 ============
 
 eli5 has basic XGBoost_ support - :func:`eli5.explain_weights`
```
Comment (kmike) [inline docs/source/autodocs/xgboost.rst]: I think 'basic' can be removed now :)

Comment (lopuhin) [inline docs/source/autodocs/xgboost.rst]: Haha, yes :)

**Comment by kmike** [comment]
Looks good!
